accepted/2020/reproducible-builds.md

```diff
@@ -0,0 +1,238 @@
+- Strong name keys are checked into source control
+
+After building and publishing the package using the latest SDK, NuGet.org validates the library and assigns a higher score.
+
+#### Developer using NuGet.org
+
+A developer is searching for a logging library on NuGet.org. They search and see two interesting libraries, one with a "verified" badge, and the other without. The developer picks the verified one, and feels confident that the package they're using does not contain code they are not aware of and have visibility to. They trust NuGet to have validated that the package meets the specified supply chain requirements.
+
+#### Corporate / Government developer at a large organization
+
+A developer in an internal team building a service needs to make sure they comply with their supply chain requirements. They use a feed containing only validated packages provided by their internal InfoSec team.
+
+#### NuGet
+
+NuGet is looking at ways to improve the security of the ecosystem and provide metrics around project/package health. Ensuring that the binaries in the project match the source is an important part, ensuring no tampering occurs in the build pipeline.
+
+NuGet will add a validation step on package ingestion:
+
+1. For each managed binary in the NuGet package, call a tool to check the status. Result is yes/no.
+2. If all of the files are successful, the package is marked as reproducible. If a package contains unmanaged binaries, then the result would be interterminite and marked accordingly.
+3. For each managed binary or PDB file in the package, add it to the NuGet symbol server. NuGet will retroactively do this step for all existing packages to-date.
+
+NuGet will allow packages that are not reproducible to be ingested; failure is not a hard block. NuGet will display the reproducible status in appropriate metadata and UI so users and tools that trust NuGet can make informed decisions without revalidating themselves.
+
+## Requirements
+
+### Goals
+
+**P0**
+
+- The validation tool must work on any managed binary, regardless of target framework
+- The validation tool must work on any operating system.
+- The validation tool must support public sources for sources and symbols.
+- The validation tool must be easy to use: single command the user invokes.
+
+**P1**
+
+- The validation tool should support non-public source and symbol locations given credentials.
+
+**P2**
+
+- The validation tool should support F#
+- The validation tool should support any .NET SDK tool which creates or modifies DLL / EXEs
+
+**Future**
+
+- Support for IL weavers
+
+### Non-Goals
+
+- Evaluating provenance of embedded resources, such as .resx or .baml.
+- Support for modification or patching during validation, to create a runnable binary that's functionally different from the original.
+- Support for exporting a rebuilt artifact
+- Supporting the tool being used to validate binaries in an environment different than what they were produced on. 
```

Comment:
What does "environment" here mean? Is it just the architecture + OS combination? Or is there more to it?

Reply:
Ideally we support all inputs which result in a deterministic build. https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/deterministic-compiler-option